### ISSUE TITLE
[L] Realtime data - pub/sub

### ISSUE THREAD

**Issue by JLWade** [bcgov/parks-data-register#370]
_As a system subscriber (A&R, DUP, bcparks.ca), I would like to have updates from the register whenever they are made and not require updates on a schedule so that I do not need to manage a schedule every night.

This ticket is to try the pub/sub method with the data register and one of the subscribers. Self serve access to be notified of a change._

The publish-subscribe (pub/sub) model enables [event-driven architecture](https://aws.amazon.com/event-driven-architecture/). Pub/sub messaging provides significant advantages to developers who build applications that rely on real-time events.

Applying this to our system will provide a number of significant benefits, eliminating polling, providing dynamic targeting among others. For example, Watching the details page of a park will allow a viewer to see the capacity levels drop in realtime as individuals book passes.  In other cases, we can alert administrators that someone else has just made a change to the record they are viewing/editing prompting them to reload or take other action (such as instant-message the person).

See https://aws.amazon.com/pub-sub-messaging/benefits/ for more information.

Note: this is in the DR repo but it is a universal product thing. How we architecturally and foundationally enable realtime data. 
Note note: there is another thing in the DR that we've left - which is the realtime pushes to other subscribers to the data register. Same same but different. Different bc when someone hits publish in the name register, we want to send out to everyone who is a consumer that it has been published.

**Acceptance Criteria:**
- [ ] building pipelines in front and and backend app

Future idea tbc by PO:
- [ ] DR - toast notification in UI saying someone has updated the system

**Comment by LindsayMacfarlane** [comment]
@marklise - would be curious to get your input on this ticket. Will/could this support any of our upcoming work...or should we icebox this for now?

**Comment by marklise** [comment]
@LindsayMacfarlane Yes this is definitely a good thing to bring in. It's not a huge amount of work and will slot in nicely for certain aspects of operating the admin side of things.

**Comment by Dianadec** [comment]
Standup notes: Overlaps with Daniel's work on scheduling - frontend subscribing to parks endpoint.

**Comment by Dianadec** [comment]
@marklise to add comment here once this is wrapped up to indicate next steps.

**Comment by marklise** [comment]
Realtime data is now enabled via the DynamoDB stream function. Every time something changes in Dynamo, the stream will send a notification to all active websockets pointing to the API gateway.  We can now use this to `push` events to browsers where we feel realtime/instant notifications are required.

Admin console showing example connect, receive message, and disconnect:

![Image](https://github.com/user-attachments/assets/97da5dce-b163-4083-a1f2-17b12b208fb6)

**Comment by LindsayMacfarlane** [comment]
Thanks for all your work on this @marklise ! Ability to access realtime data is huge. 

No testing done on my end.
